tests/phpunit/tests/test-install.php

```diff
@@ -63,20 +67,22 @@ function test_uninstall() {
 
 		update_post_meta( $item_id, '_pll_menu_item', array() );
 
+		$this->assertFalse( defined( 'PLL_REMOVE_ALL_DATA' ), 'The constant PLL_REMOVE_ALL_DATA must not be defined for this test'  );
+
 		if ( ! defined( 'WP_UNINSTALL_PLUGIN' ) ) {
 			define( 'WP_UNINSTALL_PLUGIN', true );
 		}
 
 		include_once dirname( dirname( dirname( dirname( __FILE__ ) ) ) ) . '/uninstall.php';
 		new PLL_Uninstall();
```

Comment:
Actually, the instantiation of this class as already been done in https://github.com/polylang/polylang/blob/0a50edc71c80ce29582a0991ab22013ebe6ba706/uninstall.php#L159 , that you trigger the line above.

Also, this class only contains one method, that is executed upon instantiation... That makes me believe that it is actually a script that you wrapped into a class, but you don't get OOP benefits here, and this doesn't really make the code clearer IMO.

Reply:
I did not change anything here. Classes used as namespaces. All of this has been written at at time where we had to support PHP 5.2.